openquake/hazardlib/geo/surface/planar.py

```diff
@@ -149,14 +157,25 @@ def from_corner_points(cls, mesh_spacing, top_left, top_right,
             An instance of :class:`PlanarSurface`.
         """
         strike = top_left.azimuth(top_right)
-
         dist = top_left.distance(bottom_left)
         vert_dist = bottom_left.depth - top_left.depth
         dip = numpy.degrees(numpy.arcsin(vert_dist / dist))
-
         return cls(mesh_spacing, strike, dip, top_left, top_right,
                    bottom_right, bottom_left)
 
+    @classmethod
+    def from_array(cls, mesh_spacing, array):
+        """
+        :param mesh_spacing: mesh spacing parameter
+        :param array: a composite array with fields (lon, lat, depth)
+        :returns: a :class:`PlanarSurface` instance
+        """
+        tl, tr, bl, br = _corners(array)
+        strike = tl.azimuth(tr)
+        dip = numpy.degrees(
+            numpy.arcsin((bl.depth - tl.depth) / tl.distance(bl)))
```

Comment:
`tl` and `bl` are distinct points (there are checks when creating then rupture) so their distance is nonzero.